### ISSUE TITLE
NullReferenceException in UWP app

### ISSUE THREAD

**Issue by michael-hawker** [mrlacey/Rapid-XAML-Toolkit#455]
**Describe the bug**
I unfortunately can't share too many details as it's not a piece of XAML I can share. Just noticed this notice in the Error window and thought I should log a but. At first I thought it was because I had the `rd.xml` file open from my Properties folder.

This is a compiling stable app, so everything should have been valid. If you can tell me if  there's other general information I can tell you, happy to try and take a look.

**Expected behavior**
No error message?

**Additional context**

```
Severity	Code	Description	Project	File	Line	Suppression State	Detail Description
Error	RXT999	Unexpected error occurred while parsing XAML.	HomePage.xaml	1		Please log an issue to https://github.com/mrlacey/Rapid-XAML-Toolkit/issues Reason: System.NullReferenceException: Object reference not set to an instance of an object.
   at RapidXaml.RapidXamlElement.<>c__DisplayClass39_0.<ContainsChildOrAttribute>b__2(RapidXamlElement c) in C:\Users\matt\Documents\GitHub\mrlacey-RXT\VSIX\RapidXaml.CustomAnalysis\RapidXamlElement.cs:line 302
   at System.Linq.Enumerable.Any[TSource](IEnumerable`1 source, Func`2 predicate)
   at RapidXaml.RapidXamlElement.<>c__DisplayClass39_0.<ContainsChildOrAttribute>b__1(RapidXamlAttribute a) in C:\Users\matt\Documents\GitHub\mrlacey-RXT\VSIX\RapidXaml.CustomAnalysis\RapidXamlElement.cs:line 301
   at System.Linq.Enumerable.Any[TSource](IEnumerable`1 source, Func`2 predicate)
   at RapidXaml.RapidXamlElement.ContainsChildOrAttribute(String name) in C:\Users\matt\Documents\GitHub\mrlacey-RXT\VSIX\RapidXaml.CustomAnalysis\RapidXamlElement.cs:line 297
   at RapidXaml.RapidXamlElement.ContainsDescendant(String elementName) in C:\Users\matt\Documents\GitHub\mrlacey-RXT\VSIX\RapidXaml.CustomAnalysis\RapidXamlElement.cs:line 136
   at RapidXamlToolkit.XamlAnalysis.CustomAnalysis.TwoPaneViewAnalyzer.Analyze(RapidXamlElement element, ExtraAnalysisDetails extraDetails)
   at RapidXamlToolkit.XamlAnalysis.CustomProcessorWrapper.Process(String fileName, Int32 offset, String xamlElement, String linePadding, ITextSnapshot snapshot, TagList tags, List`1 suppressions, Dictionary`2 xmlns)
   at RapidXamlToolkit.XamlAnalysis.XamlElementExtractor.Parse(String fileName, ITextSnapshot snapshot, String xaml, List`1 processors, TagList tags, List`1 suppressions, XamlElementProcessor everyElementProcessor, ILogger logger)
   at RapidXamlToolkit.XamlAnalysis.RapidXamlDocument.Create(ITextSnapshot snapshot, String fileName, IVisualStudioAbstraction vsa, String projectFile)
```

**Comment by mrlacey** [comment]
Based on the call stack I can see where this, theoretically, might happen but I can't contrive input to make it happen.
Also, based on the line number references in the output, I can't tie this to a particular build.

Do you know what version of the extension you have installed?

I'd love to know if this ever happens again.
A copy of the XAML would be really helpful if possible. Even sent in private with relevant NDAs in place ;) Or even just rename some elements in what you send if that's an option. I don't need to be able to compile it, I just need a valid XML document.

In the meantime, I'll add the version number to the above error messages to help identify the source of any future errors.

**Comment by michael-hawker** [comment]
@mrlacey I have version 0.11 of the packages, though 0.11.4 of the analysis one. Not sure which one is relevant here.

One thing that may be a pointer is that it's using a custom page class and the red squiggle is on the first character of that at the top of the document?

```xml
<pages:MySpecialPage
   ...
```

**Comment by mrlacey** [comment]
Oh no. Version 0.11 doesn't tie in with the line numbers 😕 

The red squiggle is at the top of the document as it has to go somewhere and I don't know where the problem is.
Based on the call stack I can see that the issue is with the TwoPaneView element. Or, specifically something nested two levels deep inside it. 😉 

I suppose it's good to know that it is still happening for you and wasn't just a one-off. 
That means it might be worth me digging a bit deeper to see what I can re-engineer to try and reproduce it.

**Comment by michael-hawker** [comment]
Cool, if there's an updated version I should go to then I can try updating and see if that gets you more info. Didn't think I saw a prompt in VS, but I can check again.

Here's the `TwoPaneView` snippet, it seems rather innocuous to share 🙂

```xml
        <muxc:TwoPaneView
            x:Name="TwoPaneContent"
            Grid.Column="1"
            Margin="40,0,0,0"
            MinTallModeHeight="Infinity"
            MinWideModeWidth="Infinity"
            ModeChanged="OnTwoPaneViewModeChanged">
            <muxc:TwoPaneView.Pane1>
                <!-- Grid necessary because of ContentFrame1 using deferred loaded -->
                <Grid Margin="0,32,0,0">
                    <Frame x:Name="ContentFrame1" x:Load="True" />
                </Grid>
            </muxc:TwoPaneView.Pane1>
            <muxc:TwoPaneView.Pane2>
                <!-- Grid necessary because of ContentFrame2 using deferred loaded -->
                <Grid>
                    <Frame x:Name="ContentFrame2" x:Load="False" />
                </Grid>
            </muxc:TwoPaneView.Pane2>
        </muxc:TwoPaneView>
```

**Comment by mrlacey** [comment]
Thanks for the snippet. I can now reproduce the errorI 💯

**Comment by mrlacey** [comment]
Fix will be in 0.12

**Comment by mrlacey** [comment]
This functionality has been released as part of [v 0.12.0 and is available now](https://marketplace.visualstudio.com/items?itemName=MattLaceyLtd.RapidXamlToolkit).
